src/components/ComboButton/ComboButton.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file Combo button.
+ * @copyright IBM Security 2019
+ */
+
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { Fragment, useState, useRef } from 'react';
+import { ChevronDown16, ChevronUp16 } from '@carbon/icons-react';
+import { settings } from 'carbon-components';
+
+import Button from '../Button';
+import OverflowMenu from '../OverflowMenu';
+import OverflowMenuItem from '../OverflowMenuItem';
+import { TooltipDirection } from '../IconButton/IconButton';
+
+import { namespace as buttonNamespace } from '../Button/Button';
+
+import { getComponentNamespace } from '../../globals/namespace';
+
+export const namespace = getComponentNamespace('combo-button');
+
+const { prefix } = settings;
+
+const OVERFLOW_MENU_ICON = {
+  OPEN: ChevronUp16,
+  CLOSE: ChevronDown16,
+};
+
+const ComboButton = props => {
+  const [overflowMenuIcon, setOverflowMenuIcon] = useState(
+    OVERFLOW_MENU_ICON.OPEN
+  );
+  const wrapper = useRef(null);
+  const { className, overflowMenuDirection, actions } = props;
+  const [firstAction, ...restActions] = actions;
+  const isTop = overflowMenuDirection === TooltipDirection.TOP;
+
+  const getMenuOffset = () => {
+    const { top } = wrapper.current.getBoundingClientRect();
+    return {
+      top: isTop ? top : top * -1,
+      left: 'auto',
+    };
+  };
+
+  const renderActions = items => {
+    const itemClassName = `${prefix}--list-box__menu-item`;
+    return items.map((item, index) => (
+      <OverflowMenuItem
+        className={itemClassName}
+        key={item.label}
+        itemText={
+          <Fragment>
+            <span className={`${itemClassName}-label`} title={item.label}>
+              {item.label}
+            </span>
+            <item.renderIcon className={`${itemClassName}-icon`} />
+          </Fragment>
+        }
+        primaryFocus={index === (isTop ? items.length - 1 : 0)}
+        onClick={item.onClick}
+      />
+    ));
+  };
+
+  return (
+    <div
+      className={classnames(`${namespace}__wrapper`, className)}
+      data-testid="combo-button"
+      ref={wrapper}
+      data-floating-menu-container
+    >
+      <Button
+        {...firstAction}
+        renderIcon={restActions.length > 1 ? null : firstAction.renderIcon}
+        data-testid="combo-button-primary-button"
+      >
+        {firstAction.label}
+      </Button>
+      {restActions.length > 0 && (
+        <OverflowMenu
+          menuOffset={getMenuOffset}
+          className={classnames(
+            buttonNamespace,
+            `${prefix}--btn`,
+            `${prefix}--btn--primary`
+          )}
+          menuOptionsClass={`${prefix}--list-box__menu`}
```

Comment:
I'm pretty sure this the overflow menu component already provides this classname 🤔

Reply:
I think this is the *one* class that isn't getting added. 🤔 Or at least when I tried to remove it, the class was not added (and that prevents a `maxHeight` from getting added so the menu scrolls)

Reply:
hmm that's bizarre